src/DataFramesMeta.jl

```diff
@@ -410,19 +410,39 @@ function orderby(x::AbstractDataFrame, @nospecialize(args...))
 end
 
 function orderby(x::GroupedDataFrame, @nospecialize(args...))
-
     @warn "orderby behavior now returns a `DataFrame` instead of a `GroupedDataFrame`. " *
           "Group the returned data frame to restore old behavior" maxlog = 2
 
     t = DataFrames.select(x, args...; copycols = false, keepkeys = false)
+    @assert nrow(parent(x)) == nrow(t)
     parent(x)[sortperm(t), :]
 end
 
 """
     @orderby(d, i...)
 
-Sort by criteria. When using a grouped data frame, operation
-is performed at the group level.
+Sort by criteria. When given a `GroupedDataFrame`, operation
+is performed at the group level.`@orderby` always returns a fresh
+`DataFrame` regardless of whether the input is a `DataFrame` or a
+`GroupedDataFrame`.
+
+When given a `DataFrame`, `@orderby` applies the transformation
+given by its arguments (but does not create new columns) and sorts
+the given `DataFrame` on the result, returning a new `DataFrame`.
+
+When given a `GroupedDataFrame`, `@orderby` applies the transformation
+given by its arguments *by group*. Then it returns the `parent` of the
+input `GroupedDataFrame` which has been sorted by the transformations.
```

Comment:
```suggestion
given by its arguments *by group*. Then it returns rows of the
input `GroupedDataFrame` sorted by the transformations.
```
(The parent is really an implementation detail.)